cephfs/permissions_test.go

```diff
@@ -64,3 +64,29 @@ func TestChown(t *testing.T) {
 	assert.Equal(t, uint32(sx.Uid), bob)
 	assert.Equal(t, uint32(sx.Gid), bob)
 }
+
+func TestLchown(t *testing.T) {
+	dirname := "four"
+	var bob uint32 = 1010
+	var root uint32
+
+	mount := fsConnect(t)
+	defer fsDisconnect(t, mount)
+
+	err := mount.MakeDir(dirname, 0755)
+	assert.NoError(t, err)
+	defer mount.RemoveDir(dirname)
+
+	err = mount.SyncFs()
+	assert.NoError(t, err)
+
+	err = mount.Symlink(dirname, "symlnk")
+	assert.NoError(t, err)
+	defer mount.Unlink("symlnk")
+
+	err = mount.Lchown("symlnk", bob, bob)
+	sx, err := mount.Statx("symlnk", StatxBasicStats, 0)
```

Comment:
You are getting closer but this is still not quite right.

When you run `mount.Statx("symlnk", StatxBasicStats, 0)` you get back that stat info for `dirname` which has owner root/root. This means you are not getting the actual stat of "symlnk" itself. This test passes because you haven't changed the ownership of `dirname` but it doesn't verify that the ownership of "symlnk" has changed. Try something more like this:

```golang
    sx, err := mount.Statx("symlnk", StatxBasicStats, AtSymlinkNofollow)
    assert.NoError(t, err)
    assert.Equal(t, uint32(sx.Uid), bob)
    assert.Equal(t, uint32(sx.Gid), bob)
    sx, err = mount.Statx(dirname, StatxBasicStats, AtSymlinkNofollow)
    assert.Equal(t, uint32(sx.Uid), root)
    assert.Equal(t, uint32(sx.Gid), root)
```

Note that I passed `AtSymlinkNofollow` to the 2nd call to Statx. That doesn't actually change the results of the call, because we know `dirname` is the name of a directory. But if we didn't know what dirname was passing `AtSymlinkNofollow` guarantees the API won't follow the link and then we could use the returned stat info to determine what kind of node `dirname` actually was.

Reply:
Thanks, I was confused with this part and running in circles. But this makes sense to me.